lib/widgets/selection.dart

```diff
@@ -1627,7 +1627,12 @@ class _GoToArtistSelectionAction extends StatelessWidget {
     if (content is Song) {
       HomeRouter.instance.goto(HomeRoutes.factory.content(content.getArtist()));
     } else if (content is Album) {
-      HomeRouter.instance.goto(HomeRoutes.factory.content(content.getArtist()));
+      final artist = content.getArtist();
+      if (artist != null) {
+        HomeRouter.instance.goto(HomeRoutes.factory.content(artist));
+      } else {
+        ShowFunctions.instance.showToast(msg: getl10n(controller.context).artistUnknown);
```

Comment:
Why do we event show this button then?
And can we heuristically choose the artist from the first song on album?

Reply:
Hm, I don't think we should do any heuristic, there will always be edge cases like when the first track of an album is from a collaboration or something like that. An album might have many artists or none, if we don't know about it due to missing file metadata. I think in the future we should show a dialog with all artists, but for now let's hide the button.

Reply:
I should point out that I fundamentally disagree on a premise that the artist can be null in our data model

Reply:
What do you mean by that?

If the artist can't be null, what do we do with songs that don't have the artist in the metadata and we can't resolve it online?

Reply:
> What do you mean by that?
> 
> If the artist can't be null, what do we do with songs that don't have the artist in the metadata and we can't resolve it online?

See the other thread

Reply:
> If the artist can't be null, what do we do with songs that don't have the artist in the metadata and we can't resolve it online?

Hmm, actually, I now see that in documentation artist can be null https://developer.android.com/reference/android/provider/MediaStore.Audio.AudioColumns#ARTIST_ID

> The id of the artist who created the audio file, if any

Practically speaking Android always puts there <unkown> artist ID, if the song doesn't have an artist.
But since the documentation says otherwise, let's support the case where it's null.

Sorry for the long discussion :-)